.buildkite/scripts/steps/es_snapshots/build.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .buildkite/scripts/common/util.sh
+
+echo "--- Cloning Elasticsearch and preparing workspace"
+
+cd ..
+destination="$(pwd)/es-build"
+rm -rf "$destination"
+mkdir -p "$destination"
+
+mkdir -p elasticsearch && cd elasticsearch
+
+export ELASTICSEARCH_BRANCH="${ELASTICSEARCH_BRANCH:-$BUILDKITE_BRANCH}"
+
+if [[ ! -d .git ]]; then
+  git init
+  git remote add origin https://github.com/elastic/elasticsearch.git
+fi
+git fetch origin --depth 1 "$ELASTICSEARCH_BRANCH"
+git reset --hard FETCH_HEAD
+
+ELASTICSEARCH_GIT_COMMIT="$(git rev-parse HEAD)"
+export ELASTICSEARCH_GIT_COMMIT
+
+ELASTICSEARCH_GIT_COMMIT_SHORT="$(git rev-parse --short HEAD)"
+export ELASTICSEARCH_GIT_COMMIT_SHORT
+
+# These turn off automation in the Elasticsearch repo
+export BUILD_NUMBER=""
+export JENKINS_URL=""
+export BUILD_URL=""
+export JOB_NAME=""
+export NODE_NAME=""
+export DOCKER_BUILDKIT=""
+
+# Reads the ES_BUILD_JAVA env var out of .ci/java-versions.properties and exports it
+export "$(grep '^ES_BUILD_JAVA' .ci/java-versions.properties | xargs)"
+
+export PATH="$HOME/.java/$ES_BUILD_JAVA/bin:$PATH"
+export JAVA_HOME="$HOME/.java/$ES_BUILD_JAVA"
+
+# The Elasticsearch Dockerfile needs to be built with root privileges, but Docker on our servers is running using a non-root user
+# So, let's use docker-in-docker to temporarily create a privileged docker daemon to run `docker build` on
+# We have to do this, because there's no `docker build --privileged` or similar
+
+echo "--- Setting up Docker-in-Docker for Elasticsearch"
+
+docker rm -f dind || true # If there's an old daemon running that somehow didn't get cleaned up, lets remove it first
+CERTS_DIR="$HOME/dind-certs"
+rm -rf "$CERTS_DIR"
+docker run -d --rm --privileged --name dind --userns host -p 2377:2376 -e DOCKER_TLS_CERTDIR=/certs -v "$CERTS_DIR":/certs docker:dind
```

Comment:
Is this erroring out early?
https://buildkite.com/elastic/kibana-elasticsearch-snapshot-build/builds/44#b35beddd-86ad-4b07-b9a5-0b437e81e311/789-946

Reply:
The `trap "docker rm -f dind" EXIT` failsafe that makes sure it gets cleaned up at the end is just showing that the container was already removed